src/algebra/direct_sum_graded.lean

```diff
@@ -543,6 +545,108 @@ end comm_ring
 
 end grade_zero
 
+section to_semiring
+
+variables {R : Type*} [Π i, add_comm_monoid (A i)] [add_monoid ι] [gmonoid A] [semiring R]
+variables {A}
+
+/-- If two ring homomorphisms from `⨁ i, A i` are equal on each `of A i y`,
+then they are equal.
+
+See note [partially-applied ext lemmas]. -/
+@[ext]
+lemma ring_hom_ext' (F G : (⨁ i, A i) →+* R)
+  (h : ∀ i, (F : (⨁ i, A i) →+ R).comp (of _ i) = (G : (⨁ i, A i) →+ R).comp (of _ i)) : F = G :=
+ring_hom.coe_add_monoid_hom_injective $ direct_sum.add_hom_ext' h
+
+section to_move
+
+lemma _root_.add_monoid_hom.ext_iff₂
+  {R S T : Type*} [add_zero_class R] [add_zero_class S] [add_comm_monoid T] {f g : R →+ S →+ T} :
+  f = g ↔ (∀ x y, f x y = g x y) :=
+add_monoid_hom.ext_iff.trans $ forall_congr $ λ _, add_monoid_hom.ext_iff
```

Comment:
This was convenient for proving the lemma below in term mode, but there's no prior art for it in mathlib, and the forward direction is covered by the `ext` tactic. Is it worth having despite that?

Reply:
`ext_iff` lemmas are really handy in tactic proofs when you have equality of functions as a hypothesis.

Reply:
Sure, although in a tactic proof `simp only [add_monoid_hom.ext_iff]` would achieve the same effect.